server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -712,6 +711,8 @@ public void run() {
           compactionThread.join();
           LOG.trace("Compaction thread finished.");
 
+          checkIfCanceled();
```

Comment:
The intent of doing periodic cancelation checks is to stop long running external compactions that are canceled.  If lots of quick external compaction (like 30 s) run, then a cancelation check will now be done for each of those where it was not before adding this   During the commit process atomic checks will be done, so doing the check here for every compaction does not offer a benefit over that. The compaction could still be canceled after this check before the atomic checks are done.  So maybe checking here places additional load on the metadata table and ZK w/o a clear benefit.

Thinking about only checking long running compactions, we could also make the checkIfCanceled method ignore young compactions. We currently check every 5 mins, if  a compaction is running and its age is <5min we could return and not do the check until a later call when the age is > 5min (or the check period).

Reply:
I think this was an issue with short-lived compactions. IIRC, the original issue was that @milleruntime deleted the ci table during some testing and a bunch of errors were logged and the compaction was not canceled (at least not in coordinator log).

Reply:
Would something like the following achieve the goal of less noisy logs while not always doing the check?

```suggestion
   if(err.get() != null){
           //maybe the error occured because the table was deleted or something like that, so force a cancel check
          checkIfCanceled();
   }
```